crates/trie/parallel/src/proof_task.rs

```diff
@@ -1181,78 +833,179 @@ impl ProofTaskCtx {
     }
 }
 
-/// Message used to communicate with [`ProofTaskManager`].
-#[derive(Debug)]
-pub enum ProofTaskMessage {
-    /// A request to queue a proof task.
-    QueueTask(ProofTaskKind),
-    /// A request to terminate the proof task manager.
-    Terminate,
-}
-
-/// Proof task kind.
+/// A handle that provides type-safe access to proof worker pools.
 ///
-/// When queueing a task using [`ProofTaskMessage::QueueTask`], this enum
-/// specifies the type of proof task to be executed.
-#[derive(Debug)]
-pub enum ProofTaskKind {
-    /// A storage proof request.
-    StorageProof(StorageProofInput, Sender<StorageProofResult>),
-    /// A blinded account node request.
-    BlindedAccountNode(Nibbles, Sender<TrieNodeProviderResult>),
-    /// A blinded storage node request.
-    BlindedStorageNode(B256, Nibbles, Sender<TrieNodeProviderResult>),
-    /// An account multiproof request.
-    AccountMultiproof(AccountMultiproofInput, Sender<AccountMultiproofResult>),
-}
-
-/// A handle that wraps a single proof task sender that sends a terminate message on `Drop` if the
-/// number of active handles went to zero.
-#[derive(Debug)]
+/// The handle stores direct senders to both storage and account worker pools,
+/// eliminating the need for a routing thread. All handles share reference-counted
+/// channels, and workers shut down gracefully when all handles are dropped.
+#[derive(Debug, Clone)]
 pub struct ProofTaskManagerHandle {
```

Comment:
I find this a tiny bit confusing that we have a `ProofTaskManagerHandle`, but no `ProofTaskManager`, as there's usually a pattern of having a `fn handle(&self) -> Handle` method. Not necessary to address in this PR, can be a followup.